workspaces/optic-engine/src/interactions/result.rs

```diff
@@ -17,6 +18,8 @@ pub enum InteractionDiffResult {
   // Matches
   // -------
   #[serde(skip)]
+  MatchedQueryParameters(MatchedQueryParameters),
+  #[serde(skip)]
```

Comment:
These `Matched*` types are only for internal use, so we can move on to shape diffing once there is an endpoint -> interaction match. In this case, that would mean that:

- the interaction's method and path were matched to a documented path.
- query parameters were documented for this path.
- query parameters were present on the interaction.

The latter is probably something that we'll want to revise once we get into the shape diffing logic: an interaction without query parameters could be considered as one with empty query parameters, which could indicate the documented query parameters are optional. We'll leave that until we're working on the diff flow.

Reply:
Yeah seems like a few things will need to change for the diff flow but things look good so far @niclim @JaapRood